### PR TITLE
A complete documentation for o2

### PR DESCRIPTION
**Pull request by kumarabhirup**
Here it is! I have created documentations for the o2 Library.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # What is it?
 
-O2 is a WordPress Customizer library for adding custom controls to the Customizer such as:
+o2 is a WordPress Customizer library for adding custom controls to the Customizer such as:
```
Comment (HardeepAsrani) [inline README.md]: O2 should be in capital letters.

File: README.md [hunk 2]
```diff
@@ -13,6 +13,19 @@ O2 is a WordPress Customizer library for adding custom controls to the Customize
...
+define( 'O2_DIRECTORY', get_template_directory() . '/inc/o2/' );
+define( 'O2_DIRECTORY_URI', get_template_directory_uri() . '/inc/o2/' );
+```
+- Once you add this code, you're ready to fly with o2! Now you may include the toolkits provided by this package further.
+- To learn how to install toolkits, go into the `controls` folder. There, you'll find documentations for every toolkit you need.
+
 # Contribute
 
-O2 lacks documentation. If you like the library and want to contribute, help with documentation will be appreciated.
+o2 lacks complete documentation. If you like the library and want to contribute, help with documentation will be appreciated.
```
Comment (HardeepAsrani) [inline README.md]: O2 here again.

File: controls/color-palette/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Using the Color Palette
```
Comment (HardeepAsrani) [inline controls/color-palette/README.md]: Make control title names #Color Palette Control for WordPress Customizer and similar

File: controls/color-palette/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Installing and Setting up
+
+- Hope you have downloaded and installed o2 on your theme. If you have installed that, then you may proceed for further steps.
```
Comment (HardeepAsrani) [inline controls/color-palette/README.md]: In such sections, you can just link to main readme.md files

File: controls/color-palette/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+			'label' => 'Slate',
+			'colors' => array( '#b9bbbb', '#a2a3bb', '#5e5f87', '#b3b7ee', '#fbf9ff' )
+		)
+	),
+	'priority' => 5,
+	'settings' => 'o2_color_palette'
+)));
+```
+
+For more info, you may consider this [5 minute read](http://www.hardeepasrani.com/2017/10/color-palette-control-for-wordpress-customizer/).
```
Comment (HardeepAsrani) [inline controls/color-palette/README.md]: No need to mention my site here.

Comment (HardeepAsrani) [reply controls/color-palette/README.md]: Do the same in all other readme.md files

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
@KumarAbhirup  I did a small review so can you please change it? Also, delete the package-lock.json file.

**Comment by kumarabhirup** [comment]
Yeah, made all your suggested changes. Be sure you notify if something left 👍

**Review by HardeepAsrani — APPROVED** [review]
Thanks!

**Pull request merged**